auth/src/main/java/com/firebase/ui/auth/ui/account_link/WelcomeBackPasswordPrompt.java

```diff
@@ -107,8 +107,8 @@ public void onClick(View view) {
     @Override
     protected void onActivityResult(int requestCode, int resultCode, Intent data) {
         super.onActivityResult(requestCode, resultCode, data);
-        if (requestCode == RC_CREDENTIAL_SAVE) {
-            finish(RESULT_OK, new Intent());
+        if (mSmartLock != null) {
+            mSmartLock.onActivityResult(requestCode, resultCode);
```

Comment:
I really don't like this, but I can't think of another way to do this without copying code between Activities. Thoughts?